designer/client/src/components/toolbarSettings/buttons/ActionButton.tsx

```diff
@@ -15,8 +15,18 @@ export function ActionButton({ name, type }: ActionButtonProps): JSX.Element {
     const processName = useSelector(getProcessName);
     const status = useSelector(getProcessState)?.status;
     const customActions = useSelector(getCustomActions);
+    const isDeployed = useSelector(isDeployedVersion);
 
     const action = useMemo(() => customActions.find((a) => a.name === name), [customActions, name]);
+    const disabledValue = useMemo(() => !isDeployed, [isDeployed, name]);
 
-    return action ? <CustomActionButton action={action} processName={processName} processStatus={status} type={type} /> : null;
+    return action ? (
+        <CustomActionButton
+            action={action}
+            processName={processName}
+            processStatus={status}
+            disabled={name === "run now" ? disabledValue : false}
```

Comment:
I have some concerns about changing the generic logic to handle this specific case. This approach might make the component less readable and could create challenges when new requirements arise in the future.

Would it be possible to add another CustomButton case instead of modifying the ActionButton? This could help maintain the readability and flexibility of the component. You can reference the implementation details [here](https://github.com/TouK/nussknacker/blob/d6dab4475e8af15003c9d0bb467d2e27094c1820/designer/client/src/components/toolbarSettings/buttons/TOOLBAR_BUTTONS_MAP.ts#L43-L42).

Reply:
Moved this change to https://github.com/TouK/nussknacker/pull/6266